packages/sdk/src/sdk.ts

```diff
@@ -983,7 +939,15 @@ export class NxtpSdk {
     timeout?: number,
   ): void {
     const args = [timeout, callback].filter((x) => !!x);
-    this.evts[event].pipe(filter).attachOnce(...(args as [number, any]));
+    if (
+      [NxtpSdkEvents.SenderTransactionPrepared, NxtpSdkEvents.ReceiverTransactionPrepared].includes(
+        event as "SenderTransactionPrepared" | "ReceiverTransactionPrepared", // so stupid...
```

Comment:
why is this needed? lol

Reply:
Because the `NxtpSdkEvents` has more than just those lol.